cmake/patches/protobuf/protobuf_cmake.patch

```diff
@@ -12,11 +12,12 @@ index 04cb3303a..4025805cf 100644
      /wd4305 # 'identifier' : truncation from 'type1' to 'type2'
      /wd4307 # 'operator' : integral constant overflow
      /wd4309 # 'conversion' : truncation of constant value
-@@ -259,7 +257,6 @@ if (MSVC)
+@@ -259,7 +257,7 @@ if (MSVC)
      /wd4355 # 'this' : used in base member initializer list
      /wd4506 # no definition for inline function 'function'
      /wd4800 # 'type' : forcing value to bool 'true' or 'false' (performance warning)
 -    /wd4996 # The compiler encountered a deprecated declaration.
++    ${onnxruntime_PROTOBUF_EXTRA_WARNING_DISABLEMENT}
```

Comment:
I feel it is a bit complicated. We created a patch for not disabling the warnings, then we create another patch on the patch to enable the warnings again. It should be simpler if we just skip this patch?

Reply:
I thought you do not want the warnings to be disabled.  That's why you added onnxruntime_PROTOBUF_EXTRA_WARNING_DISABLEMENT. So, what's the purpose ? You want to disable more warnings or you want to not disable some of the warnings (that are listed here in this file)?

Reply:
To clarify, we are talking about how to compile protobuf library, Not ORT's code.